src/LIBSVM.jl

```diff
@@ -363,9 +381,57 @@ function svmpredict{T, U<:Real}(model::SVMModel{T},
     for i = 1:ninstances
         output = ccall(fn, Float64, (Ptr{Void}, Ptr{SVMNode}, Ptr{Float64}),
             model.ptr, nodeptrs[i], pointer(decvalues, nlabels*(i-1)+1))
-        class[i] = model.labels[round(Int,output)]
+        if model.param[1].svm_type==Int32(2) # if one class SVM
+          class[i]=output
+        else
+          class[i] = model.labels[round(Int, output)]
+        end
     end
 
     (class, decvalues)
 end
+
+function svmpredict!{T, U<:Real}(svmpredict_out::Tuple{Array{Any,1},Array{Float64,2},Array{LIBSVM.SVMNode,2},Array{Ptr{LIBSVM.SVMNode},1}}
+                                 , model::SVMModel{T}, instances::AbstractMatrix{U})
+  (class, decvalues, nodes, nodeptrs) = svmpredict_out
+  @assert size(class, 1) == size(decvalues, 2)
+    global verbosity
+    ninstances = size(instances, 2)
+
+    if size(instances, 1) != model.nfeatures
+        error("Model has $(model.nfeatures) but $(size(instances, 1)) provided")
+    end
+
+    instances2nodes!(nodes, nodeptrs, instances)
+    #class = Array(T, ninstances)
+    nlabels = length(model.labels)
+    #decvalues = Array(Float64, nlabels, ninstances)
+
+    verbosity = model.verbose
+    fn = model.param[1].probability == 1 ? svm_predict_probability() :
+        svm_predict_values()
+    for i = 1:ninstances
+        output = ccall(fn, Float64, (Ptr{Void}, Ptr{SVMNode}, Ptr{Float64}),
+            model.ptr, nodeptrs[i], pointer(decvalues, nlabels*(i-1)+1))
+        if model.param[1].svm_type==Int32(2) # if one class SVM
+          class[i]=output
+        else
+          class[i] = model.labels[round(Int, output)]
+        end
+    end
+
+    (class, decvalues)
+end
```

Comment:
Dear Simonster,

I adressed your issues now and the runtests are working. Can you merge the pull request into your master branch ?

Many Thanks,
Milan


> On 03 Sep 2016, at 20:34, Simon Kornblith <notifications@github.com> wrote:
> 
> In src/LIBSVM.jl <https://github.com/simonster/LIBSVM.jl/pull/17#discussion_r77441160>:
> 
> > +
> > +    verbosity = model.verbose
> > +    fn = model.param[1].probability == 1 ? svm_predict_probability() :
> > +        svm_predict_values()
> > +    for i = 1:ninstances
> > +        output = ccall(fn, Float64, (Ptr{Void}, Ptr{SVMNode}, Ptr{Float64}),
> > +            model.ptr, nodeptrs[i], pointer(decvalues, nlabels*(i-1)+1))
> > +        if model.param[1].svm_type==Int32(2) # if one class SVM
> > +          class[i]=output
> > +        else
> > +          class[i] = model.labels[round(Int, output)]
> > +        end
> > +    end
> > +
> > +    (class, decvalues)
> > +end
> Likewise, better to avoid duplicating this code twice and instead define svmpredict to call svmpredict!.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/simonster/LIBSVM.jl/pull/17/files/a4a283bbe68b2097d0d442926586b4b68df7708b#r77441160>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AQftFZDYIzmT1Zp87qLb2FmxpQE8y7Hlks5qmb2kgaJpZM4JruI9>.
>